internal/app/synchronizer/evaluator_client.go

```diff
@@ -103,56 +122,115 @@ func (ec *evaluatorClient) initialize() error {
 	return nil
 }
 
-func (ec *evaluatorClient) grpcEvaluate(ctx context.Context, proposals []*pb.Match) (results []*pb.Match, err error) {
-	resp, err := ec.grpcClient.Evaluate(ctx, &pb.EvaluateRequest{Matches: proposals})
-	if err != nil {
-		return nil, err
+func (ec *evaluatorClient) grpcEvaluate(ctx context.Context, proposals []*pb.Match) ([]*pb.Match, error) {
+	stream, err := ec.grpcClient.Evaluate(ctx)
+
+	waitc := make(chan evaluatorResult)
+	defer close(waitc)
+	go func() {
+		results := []*pb.Match{}
+		for {
+			resp, recvErr := stream.Recv()
+			if recvErr == io.EOF {
+				// read done.
+				waitc <- evaluatorResult{results: results, err: nil}
+				return
+			}
+			if recvErr != nil {
+				waitc <- evaluatorResult{results: nil, err: recvErr}
+				return
+			}
+			results = append(results, resp.GetMatch())
+		}
+	}()
+
+	for _, proposal := range proposals {
+		if err = stream.Send(&pb.EvaluateRequest{Match: proposal}); err != nil {
+			return nil, err
+		}
+	}
+
+	if err = stream.CloseSend(); err != nil {
+		logger.Errorf("failed to close the send stream: %s", err.Error())
 	}
 
-	return resp.GetMatches(), nil
+	result := <-waitc
+	return result.results, result.err
 }
 
 func (ec *evaluatorClient) httpEvaluate(ctx context.Context, proposals []*pb.Match) (results []*pb.Match, err error) {
```

Comment:
So looks like this one does a lot more than Evaluator test? :) I guess its not ready for review yet.

Reply:
Yup, that's why it has a `dependency-on-pending-pr` tag :p. Ready for review now!